sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -0,0 +1,199 @@
+ */
+public final class DefaultRedirectStrategy implements RedirectStrategy {
+    private final ClientLogger logger = new ClientLogger(DefaultRedirectStrategy.class);
+
+    private static final int DEFAULT_MAX_REDIRECT_ATTEMPTS = 3;
+    private static final String DEFAULT_REDIRECT_LOCATION_HEADER_NAME = "Location";
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    private static final int TEMPORARY_REDIRECT_STATUS_CODE = 307;
+    private static final Set<HttpMethod> DEFAULT_REDIRECT_ALLOWED_METHODS = new HashSet<HttpMethod>() {
+        {
+            add(HttpMethod.GET);
+            add(HttpMethod.HEAD);
+        }
+    };
+
+    private final int maxAttempts;
+    private final String locationHeader;
+    private final Set<HttpMethod> redirectMethods;
+
+    /**
+     * Creates an instance of {@link DefaultRedirectStrategy} with a maximum number of redirect attempts 3,
+     * header name "Location" to locate the redirect url in the response headers and {@link HttpMethod#GET}
+     * and {@link HttpMethod#HEAD} as allowed methods for performing the redirect.
+     */
+    public DefaultRedirectStrategy() {
+        this(DEFAULT_MAX_REDIRECT_ATTEMPTS, DEFAULT_REDIRECT_LOCATION_HEADER_NAME, DEFAULT_REDIRECT_ALLOWED_METHODS);
+    }
+
+    /**
+     * Creates an instance of {@link DefaultRedirectStrategy} with the provided number of redirect attempts and
+     * default header name "Location" to locate the redirect url in the response headers and {@link HttpMethod#GET}
+     * and {@link HttpMethod#HEAD} as allowed methods for performing the redirect.
+     *
+     * @param maxAttempts The max number of redirect attempts that can be made.
+     * @throws IllegalArgumentException if {@code maxAttempts} is less than 0.
+     */
+    public DefaultRedirectStrategy(int maxAttempts) {
+        this(maxAttempts, DEFAULT_REDIRECT_LOCATION_HEADER_NAME, DEFAULT_REDIRECT_ALLOWED_METHODS);
+    }
+
+    /**
+     * Creates an instance of {@link DefaultRedirectStrategy}.
+     *
+     * @param maxAttempts The max number of redirect attempts that can be made.
+     * @param locationHeader The header name containing the redirect URL.
+     * @param allowedMethods The set of {@link HttpMethod} that are allowed to be redirected.
+     * @throws IllegalArgumentException if {@code maxAttempts} is less than 0.
+     */
+    public DefaultRedirectStrategy(int maxAttempts, String locationHeader, Set<HttpMethod> allowedMethods) {
+        if (maxAttempts < 0) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Max attempts cannot be less than 0."));
+        }
+        this.maxAttempts = maxAttempts;
+        this.locationHeader = locationHeader == null ? DEFAULT_REDIRECT_LOCATION_HEADER_NAME : locationHeader;
+        this.redirectMethods = allowedMethods == null ? DEFAULT_REDIRECT_ALLOWED_METHODS : allowedMethods;
```

Comment:
We should use default values if `locationHeader` and `allowedMethods` are null or *empty* and document it.